### PR TITLE
fix(suite-native): update QR screen brightness on going to bg 

### LINKED ISSUE #8688: Receive screen brightness is not getting back to previous settings
Steps to reproduce:
Set the brightness of the device low
Go to the app and generate receive address
Screen brightness up
Shut out the app
Brightness is still up

Expected behaviour: Brightness comes back to initial settings.

Video:
https://github.com/trezor/trezor-suite/assets/84183817/702d8a44-25aa-4137-a303-2c5add1777f5

### PR DESCRIPTION
**Pull request by vytick**
## Description

Changes brightness when app goes to background and back while showing receive QR code

## Related Issue

resolves #8688

### CODE REVIEW

File: suite-native/qr-code/src/components/QRCode.tsx [hunk 1]
```diff
@@ -32,6 +32,25 @@ export const QRCode = ({ data }: QRCodeProps) => {
     const { applyStyle } = useNativeStyles();
     const [originalBrightnessValue, setOriginalBrightnessValue] = useState<number>();
 
+    useEffect(() => {
+        const subscription = AppState.addEventListener('change', nextAppState => {
+            if (!originalBrightnessValue) return;
+
+            // When app goes to background, restore the original brightness value.
+            if (nextAppState === 'inactive') {
```
Comment (PeKne) [inline suite-native/qr-code/src/components/QRCode.tsx]:
```suggestion
            if (nextAppState === 'background') {
```
The `inactive` state is supported only by iOS. More here: https://reactnative.dev/docs/appstate#app-states.

File: suite-native/qr-code/src/components/QRCode.tsx [hunk 2]
```diff
@@ -32,6 +32,25 @@ export const QRCode = ({ data }: QRCodeProps) => {
...
+                Brightness.setBrightnessAsync(originalBrightnessValue);
+            }
+            // When app goes to foreground, store the original brightness value and set full.
```
Comment (PeKne) [inline suite-native/qr-code/src/components/QRCode.tsx]: The comment is wrong, you are not storing anything here.

**Review by PeKne — CHANGES_REQUESTED** [review]

File: suite-native/qr-code/src/components/QRCode.tsx (hunk 1 above)
Comment (vytick) [reply suite-native/qr-code/src/components/QRCode.tsx]: good point, thnks

File: suite-native/qr-code/src/components/QRCode.tsx (hunk 2 above)
Comment (vytick) [reply suite-native/qr-code/src/components/QRCode.tsx]: nice catch

Comment (vytick) [reply suite-native/qr-code/src/components/QRCode.tsx]: cfb332afa

File: suite-native/qr-code/src/components/QRCode.tsx (hunk 1 above)
Comment (vytick) [reply suite-native/qr-code/src/components/QRCode.tsx]: cfb332afa

**Review by PeKne — APPROVED** [review]

**Pull request merged**